edge-modules/DirectMethodSender/Program.cs

```diff
@@ -24,21 +26,47 @@ public static async Task<int> MainAsync()
             (CancellationTokenSource cts, ManualResetEventSlim completed, Option<object> handler) = ShutdownHandler.Init(TimeSpan.FromSeconds(5), Logger);
             DirectMethodSenderBase directMethodClient = null;
             ModuleClient reportClient = null;
+            Option<Uri> analyzerUrl = Settings.Current.AnalyzerUrl;
+            Option<Uri> testReportCoordinatorUrl = Settings.Current.TestResultCoordinatorUrl;
+
             try
             {
+                Guid batchId = Guid.NewGuid();
+                Logger.LogInformation($"Batch Id={batchId}");
```

Comment:
If you introduce batch id, please have this value sent to receiver, so it should have enough data to report to TRC.

Reply:
It is done below. Line:58-68

Reply:
I am talking about when receiving on DM receiver; not in DM sender.

Reply:
Oh I see. I created a task for it.